models/core/strategy_manager.py

```diff
@@ -43,7 +43,22 @@ def _load_strategies(self) -> Dict[str, Dict[str, Any]]:
         try:
             with open(self.strategies_file) as f:
                 data = yaml.safe_load(f) or {}
-                return data.get("strategies", {})
+                strategies_list = data.get("strategies", [])
+                
+                # Strategies must be an array
+                if not isinstance(strategies_list, list):
```

Comment:
We should probably just make sure any files we include are already arrays and if we ever see one that's not, throw an error.